### PR TITLE
First pass at creating a basic todo app as a complete example.

### LINKED ISSUE #33: Include runnable example in source code?
I've managed to successfully migrate a mobx v5  and mobx-react project over to mobx-react-lite, and was pleasantly surprisedly how relatively easy it was.

That being said, I can see the process being daunting for people that aren't as familiar with functional composition, use of context, etc.

@FredyC is there any desire to have a runnable example project? If so, I will happily volunteer to create something that can serve as way to complement the existing documentation in the readme.

### PR DESCRIPTION
**Pull request by larrymyers**
This still needs some polish and isn't ready for merge yet. Wanted to get an early PR created to get some feedback.

@FredyC is this something you think will be valuable to include directly in the project?

Closes #33

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,12 @@
 /node_modules/
 
+/example/node_modules/
```
Comment (FredyC) [inline .gitignore]: I don't think you need these two, it should work recursively. Unless you have some IDE that would have a problem with it? In that case, I would rather create separate `.gitignore` in an `example` folder.

File: example/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "mobx-react-lite-example",
+    "version": "0.0.1",
+    "main": "index.js",
```
Comment (FredyC) [inline example/package.json]: I don't think there should be a `main` here as it's not a module to be consumed.

File: example/src/todoForm.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    const todoStore = useContext(TodoStoreContext);
+    const input = React.createRef<HTMLInputElement>();
+
+    useEffect(() => {
+        if (input.current) {
+            input.current.focus();
+        }
+    });
+
+    const addTodo = (evt: FormEvent) => {
```
Comment (FredyC) [inline example/src/todoForm.tsx]: What about `useCallback` here?

File: .gitignore (hunk 1 above)
Comment (larrymyers) [reply .gitignore]: the leading leading slash only matches root of the project, so just the top level node_modules

File: example/package.json (hunk 2 above)
Comment (larrymyers) [reply example/package.json]: yeah, I'll trim that out, just came with the "init" command from yarn.

File: example/src/todoForm.tsx (hunk 3 above)
Comment (larrymyers) [reply example/src/todoForm.tsx]: Good idea. I'm still working my way through all the new hooks included in React out of the box.

**Comment by larrymyers** [comment]
@FredyC Thanks for the feedback. I'll go ahead and make more revisions to include more of the API in a way that makes sense so the example provides more value to new users of the project.

File: .gitignore (hunk 1 above)
Comment (FredyC) [reply .gitignore]: Alright, missed that one, let's change it to `node_modules` then, I don't see a reason why not.

**Comment by RayLuxembourg** [comment]
I have tried to implement this code, but change detection have not been triggered.

**Comment by larrymyers** [comment]
@FredyC I'm going to close this PR. I'm not making progress on it and it looks like most people are looking for smaller, more concise examples anyway.

**Pull request closed without merging**